xrpl/utils/txn_parser/utils/transaction.py

```diff
@@ -0,0 +1,183 @@
+"""Helper functions to format raw transaction data."""
+
+from typing import Any, Dict, List, Union, cast
+
+from pydash import map_  # type: ignore
+from typing_extensions import Literal
+
+from xrpl.utils.txn_parser.utils.types import (
+    AccountBalance,
+    NormalizedFields,
+    NormalizedNode,
+    RawTxnType,
+    SubscriptionRawTxnType,
+    XRPLTxnFieldsException,
+)
+
+
+def validate_transaction_fields(
+    transaction_data: Union[RawTxnType, SubscriptionRawTxnType],
+) -> None:
+    """
+    Check if the transaction fields are valid.
+
+    Args:
+        transaction_data: The raw transaction data.
+
+    Raises:
+        XRPLTxnFieldsException: If the raw transaction data is malformed.
+    """
+    if "meta" not in transaction_data:
+        raise XRPLTxnFieldsException(
+            "Malformed transaction fields: Transaction field 'meta' must be included."
+        )
+    meta = transaction_data["meta"]
+    if "AffectedNodes" not in meta or not meta["AffectedNodes"]:
+        raise XRPLTxnFieldsException("Malformed transaction fields: No nodes provided.")
+
+
+def normalize_transaction(
+    transaction_data: SubscriptionRawTxnType,
+) -> RawTxnType:
+    """
+    Formats the raw transaction data into one standard format.
```

Comment:
```suggestion
    Formats the raw transaction data from a `transaction` subscription to match what is returned from a `tx` request.
```
Feel free to change the wording of this